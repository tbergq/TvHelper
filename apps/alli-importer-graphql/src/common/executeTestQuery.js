// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
