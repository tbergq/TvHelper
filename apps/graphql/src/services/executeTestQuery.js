// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
