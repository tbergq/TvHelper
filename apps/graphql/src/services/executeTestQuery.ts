import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
