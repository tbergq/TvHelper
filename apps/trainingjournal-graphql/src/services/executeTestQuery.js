// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
