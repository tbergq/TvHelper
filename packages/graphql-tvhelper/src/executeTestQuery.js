// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
