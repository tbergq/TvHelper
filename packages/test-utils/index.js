// @flow

export { default as withMongoDb } from './src/useMongoDb';
