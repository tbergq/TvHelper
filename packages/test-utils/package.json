{
  "name": "@tbergq/test-utils",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "mongodb-memory-server": "^5.2.0",
    "mongoose": "^5.6.9"
  }
}
