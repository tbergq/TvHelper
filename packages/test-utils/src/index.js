// @flow

export { screen, fireEvent, act, waitFor } from '@testing-library/react';
export { default as render } from './render';
export { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
