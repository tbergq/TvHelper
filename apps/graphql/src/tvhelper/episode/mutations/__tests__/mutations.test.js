// @flow

import { generateTestsFromFixtures } from '@adeira/test-utils';
import {
  UserRepository,
  tvHelperConnection,
  WatchedEpisodeRepository,
} from '@tbergq/tvhelper-persistence';

import executeTestQuery from '../../../../services/executeTestQuery';

let userId;

describe('episode mutations', () => {
  beforeEach(async () => {
    const user = await UserRepository.createUser({
      password: 'lol',
      username: 'lol',
      email: 'lol',
    });
    userId = user.id;
    await WatchedEpisodeRepository.markAsWatched(user.id, 6);
  });

  afterEach(async () => {
    await tvHelperConnection.collection('users').drop();
    await tvHelperConnection.collection('watchedepisodes').drop();
  });

  generateTestsFromFixtures(`${__dirname}/__fixtures__`, input =>
    executeTestQuery(input, null, { user: { id: userId } }),
  );
});
