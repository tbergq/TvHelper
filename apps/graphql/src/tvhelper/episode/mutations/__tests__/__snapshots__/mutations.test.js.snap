// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`episode mutations matches expected output: DeleteWatchedEpisode.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation {
  deleteWatchedEpisode(episodeId: "ZXBpc29kZTo2") {
    success
    episode {
      id
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "deleteWatchedEpisode": {
      "success": true,
      "episode": {
        "id": "RXBpc29kZTo2"
      }
    }
  }
}
`;

exports[`episode mutations matches expected output: MarkAsWatched.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation {
  markAsWatched(episodeId: "ZXBpc29kZTo2") {
    success
    episode {
      id
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "markAsWatched": {
      "success": true,
      "episode": {
        "id": "RXBpc29kZTo2"
      }
    }
  }
}
`;
